src/sources/envoy_als/mod.rs

```diff
@@ -0,0 +1,92 @@
+mod grpc;
+#[cfg(all(test, feature = "envoy-als-integration-tests"))]
+mod integration_tests;
+#[cfg(test)]
+mod tests;
+
+use self::grpc::Service;
+use crate::{
+    config::{DataType, GenerateConfig, Output, Resource, SourceConfig, SourceContext},
+    sources::{util::grpc::run_grpc_server, Source},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+use envoy_proto::envoy::service::accesslog::v3::access_log_service_server::AccessLogServiceServer;
+use futures::{FutureExt, TryFutureExt};
+use std::net::SocketAddr;
+use vector_common::internal_event::EventsReceived;
+use vector_config::configurable_component;
+use vector_core::config::LogNamespace;
+
+/// Configuration for the `envoy_als` source.
+#[configurable_component(source("envoy_als"))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct EnvoyAlsConfig {
+    #[configurable(derived)]
+    grpc: GrpcConfig,
+}
+
+/// Configuration for the `envoy_als` gRPC server.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+struct GrpcConfig {
+    /// The address to listen for connections on.
+    ///
+    /// It _must_ include a port.
+    address: SocketAddr,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    tls: Option<TlsEnableableConfig>,
+}
+
+impl GenerateConfig for EnvoyAlsConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            grpc: GrpcConfig {
+                address: "0.0.0.0:9999".parse().unwrap(),
+                tls: Default::default(),
+            },
+        })
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+impl SourceConfig for EnvoyAlsConfig {
+    async fn build(&self, cx: SourceContext) -> crate::Result<Source> {
+        let events_received = register!(EventsReceived);
+
+        let grpc_tls_settings = MaybeTlsSettings::from_config(&self.grpc.tls, true)?;
+        let grpc_service = AccessLogServiceServer::new(Service {
+            events_received: events_received.clone(),
+            pipeline: cx.out.clone(),
+            shutdown: cx.shutdown.clone(),
+        })
+        .accept_compressed(tonic::codec::CompressionEncoding::Gzip);
+        let grpc_source = run_grpc_server(
+            self.grpc.address,
+            grpc_tls_settings,
+            grpc_service,
+            cx.shutdown.clone(),
+        )
+        .map_err(|error| {
+            error!(message = "Source future failed.", %error);
+        });
+
+        Ok(grpc_source.boxed())
+    }
+
+    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
+        vec![Output::default(DataType::Log)]
+    }
```

Comment:
Flagging this as a note to myself and other reviewers of the core Vector team- this source will need to have log namespacing support added. If not in this PR, then quickly after this PR is merged in.

Reply:
A heads up @wez470  - discussed this internally , and we have an in progress effort for having proper log namespacing in vector's components. We just recently finished adding this to all existing sources, and are about to do the same for sinks.
As of yet, we don't have documentation on how to do this for contributors like yourself who are likely unfamiliar with what is required.
As such, @StephenWakely will be preparing said documentation for you to follow in this PR.

In the meantime to get an idea, you can take a look at `sources/dnstap/schema.rs` , and these GH items:

RFC (background context in case you're curious about the motivation) : https://github.com/vectordotdev/vector/blob/edacee2482be1e79e10ea9a3d6766114cae44e67/rfcs/2022-04-20-12187-log-namespacing.md

Example source adding proper log namespacing, including general steps: https://github.com/vectordotdev/vector/pull/13720

Reply:
Sounds good.

Reply:
The log namespacing tutorial has landed: https://github.com/vectordotdev/vector/blob/master/docs/tutorials/lognamespacing.md

Thanks @StephenWakely  for putting that together.

This is unfortunately not a small set of changes we're asking for, but it's mainly because of all the custom fields.
The actual changes to add log namespacing aren't complex per se, it's mostly about understanding what to do and where, which is where the tutorial should help out.

The resources I linked in my earlier comment are still good to review as well.

Let us know if you have questions~